taraxa/state/dpos_2.0/precompiled/data_types/data_types.go

```diff
@@ -0,0 +1,113 @@
+package data_types
+
+import (
+	"math/big"
+
+	"github.com/Taraxa-project/taraxa-evm/common"
+	"github.com/Taraxa-project/taraxa-evm/taraxa/util/keccak256"
+)
+
+type StorageData = map[*common.Hash][]byte
+
+// Validator basic info
+type ValidatorBasicInfo struct {
+	// TotalStake == sum of all delegated tokens to the validator
+	TotalStake *big.Int
+
+	// Commission
+	Commission *big.Int
+
+	// Rewards accumulated from delegators rewards based on commission
+	CommissionRewards *big.Int
+
+	// // Short description
+	// // TODO: optional - maybe we dont want this ?
+	// Description string
+
+	// // Validator's website url
+	// // TODO: optional - maybe we dont want this ?
+	// Endpoint string
+}
+
+func (vb ValidatorBasicInfo) Serialize(pos *common.Hash, out StorageData) {
+	out[pos] = vb.TotalStake.Bytes()
+	pos.Inc()
```

Comment:
Seems to me that each Inc() usage creates possible hash collisions, which is most likely not going to happen, but why would we use such design when it is not necessary ?